pgjdbc/src/test/java/org/postgresql/jdbc/TimeZoneTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2025, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.jdbc;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.postgresql.test.DateTimeZoneStamp;
+import org.postgresql.test.SlowTests;
+import org.postgresql.test.TestUtil;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TimeZone;
+
+/**
+ * TimeZone is a global parameter affecting all APIs used by an application. User must be able
+ * to choose any available TimeZone or use system default. In any case application must
+ * connect to database successfully and must not misinterpret time-related data. This test
+ * is unlikely to catch an error when executed in local environment, where both sides,
+ * client and server, share the same OS and locale-related software components. But it highlights
+ * perfectly the problem when software relies on similarity of independently managed clients and server.
+ */
+@RunWith(Parameterized.class)
+public class TimeZoneTest {
+  @Nullable
+  private static TimeZone savedTimeZone = null;
```

Comment:
```suggestion
  private @Nullable TimeZone savedTimeZone = null;
```

`@Nullable` belongs to a type which is `TimeZone` rather than to a field